### PR TITLE
feat: setWindowButtonPosition based on zoom factor for macOS

### PR DESCRIPTION
**Pull request by darwinx64**
https://github.com/user-attachments/assets/19b862f0-6a64-4ef2-a058-cc6fe1c97d4a

### CODE REVIEW

File: src/main/mainWindow.ts [hunk 1]
```diff
@@ -235,14 +244,67 @@ function initMenuBar(win: BrowserWindow) {
             click() {
                 app.quit();
             }
+        }
+    ] satisfies MenuItemList;
+    const viewSubMenu = [
```
Comment (Vendicated) [inline src/main/mainWindow.ts:249]: whats this

Comment (darwinx64) [reply src/main/mainWindow.ts:249]: What u mean

Comment (Vendicated) [reply src/main/mainWindow.ts:249]: why are you making a new menu unrelated to the feature here

Comment (darwinx64) [reply src/main/mainWindow.ts:249]: This is the only way of detecting zoom changing as far as I know
I believe there's like window.webContents.on("zoom-change"), but it just doesn't work at all or it's for something else like trackpad zooming but im not sure 
You could detect Cmd + plus/minus but this wouldn't detect when zoom is changed from menubar but if you add a new menu it detects both that and change in menubar so it seems to be the all around best solution

**Pull request closed without merging**